linters/translation_filter_scanner.js

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+const PIPE_T_RXP = /\{\{\s*('|")((?:(?!\1).)+)\1\s*\|\s*t(?:ranslate)?(?::(.*?))?\s*(?:\||\}\})/g;
+
+module.exports = class TranslationFilterScanner {
```

Comment:
Refactored this out to make it easier to test in isolation and so the `I18nLinter` doesn't have to care about how we're actually finding theses (so we can potentially refactor this to something a bit more maintainable than this inscrutable regex).